src/settings.c

```diff
@@ -474,6 +474,17 @@ void load_settings(char *cmdline_config_path)
                 "Scale larger icons down to this size, set to 0 to disable"
         );
 
+        // restrict the icon size to a reasonable limit if we have a fixed width.
+        // Otherwise the layout will be broken by too large icons.
+        // See https://github.com/dunst-project/dunst/issues/540
+        if (settings.geometry.width_set) {
+                int icon_size_limit = settings.geometry.w / 2;
+                if (settings.max_icon_size == 0 ||
+                    settings.max_icon_size > icon_size_limit) {
+                        settings.max_icon_size = icon_size_limit;
```

Comment:
For consistency: In the past we've put the logic concatenators of if conditions in front of the new line: 

```
                if (   settings.max_icon_size == 0
                    || settings.max_icon_size > icon_size_limit) {
                        settings.max_icon_size = icon_size_limit;
```